### PR TITLE
ci: add style check for doc comments

### PR DESCRIPTION
**Pull request by Akmadan23**
Discussed in https://github.com/nvim-tree/nvim-tree.lua/pull/2555#issuecomment-1826209931

### CODE REVIEW

File: scripts/doc-comments.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (alex-courtis) [inline scripts/doc-comments.sh:1]: I always try to use `/bin/sh` after I caused a shell compatibility production incident 20 years ago.

However, we need bash features here and we know that CI provides bash so we are OK.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Fails nicely.

shellcheck :+1: 

We're going to need to fix all these before we can merge: `find lua -type f -exec sed -i 's/^--- @/---@/g' {} \;`

File: scripts/doc-comments.sh (hunk 1 above)
Comment (gegoune) [reply scripts/doc-comments.sh:1]: If you insist on `sh` (it's fine either way):

```sh
#!/bin/sh

out=$(grep -nr "^--- @" lua)

if [ "$out" ]; then
	last_file=""
	printf %s "$out" | while read -r line; do
		file="$(echo "$line" | cut -d: -f1)"
		if [ "$file" != "$last_file" ]; then
			echo "$file:" >&2
			last_file="$file"
		fi
		echo "$line" | awk -F: '{ printf("  line %s: %s\n", $2, $3) }' >&2
	done
	exit 1
fi
```

Comment (alex-courtis) [reply scripts/doc-comments.sh:1]: Works nicely, shellcheck is happy.

I will sleep better at night.

**Review by alex-courtis — APPROVED** [review]
CI all passing now

**Pull request merged**